packages/discord.js/typings/index.d.ts

```diff
@@ -579,7 +584,15 @@ export abstract class CommandInteraction<Cached extends CacheType = CacheType> e
       | JSONEncodable<APIModalInteractionResponseCallbackData>
       | ModalComponentData
       | APIModalInteractionResponseCallbackData,
-  ): Promise<void>;
+    options?: ShowModalOptions,
+  ): Promise<undefined>;
+  public showModal(
+    modal:
+      | JSONEncodable<APIModalInteractionResponseCallbackData>
+      | ModalComponentData
+      | APIModalInteractionResponseCallbackData,
+    options: ShowModalOptions & { withResponse: true },
+  ): Promise<InteractionCallbackResponse>;
```

Comment:
This overload will never be encountered as the overload above satisfies all inputs. It should be swapped.

Also, I suppose it would be ideal to write tests for these overloads.